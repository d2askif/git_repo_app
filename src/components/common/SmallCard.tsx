import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
interface Props {
  title: string;
  icon: string;
  label: string;
}

const styles = StyleSheet.create({
  container: {},
  label_container: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    color: '#6D7195',
    opacity: 0.8,
  },
  label: {
    marginLeft: 4,
    color: '#999BB1',
  },
});
const SmallCard = ({title, icon, label}: Props) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <View style={styles.label_container}>
        <Icon name={icon} size={15} color="#D7BFB6" />
        <Text style={styles.label}>{label}</Text>
      </View>
    </View>
  );
};

export default SmallCard;
