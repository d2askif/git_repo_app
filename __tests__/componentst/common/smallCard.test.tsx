import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import SmallCard from '../../../src/components/common/SmallCard';

test('SmallCard', () => {
  const props = {
    title: 'title',
    label: 'label',
    icon: 'star',
  };
  const snapshot = renderer.create(<SmallCard {...props} />);
  expect(snapshot).toMatchSnapshot();
});
