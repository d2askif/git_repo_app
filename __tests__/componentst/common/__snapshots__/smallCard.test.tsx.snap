// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallCard 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "color": "#6D7195",
        "fontSize": 20,
        "opacity": 0.8,
      }
    }
  >
    title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#D7BFB6",
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï€…
    </Text>
    <Text
      style={
        Object {
          "color": "#999BB1",
          "marginLeft": 4,
        }
      }
    >
      label
    </Text>
  </View>
</View>
`;
